storybookjs/storybook
# Core: Restore `/preview` etc package exports; return unresolved path from presets.

Issue:

Chromatic regressions introduced in https://github.com/storybookjs/storybook/pull/19007, ultimately due to many addons not working in example apps.

The root issue that was being worked around was that if a package has an `exports` field, Vite can only import an entry there. i.e. it can `import('@storybook/addon-backgrounds/preview')` not `import('@storybook/addon-backgrounds/dist/preview.js')`, which is what the presets code was resolving it to.

## What I did

In the presets code, try resolving an import path, but return the import path unmodified, so the builder can import it itself.

NOTE: This means that import paths (e.g. `previewEntries`) will be of the form `'@storybook/addon-backgrounds/preview'` rather than `/Users/xyz/..../node_modules/@storybook/addon-backgrounds/preview`. Is this a problem @ndelangen? If so we could "half resolve" the import (ie. absolutize the import path).

## How to test

- Run example app (in both Vite and Webpack), check the addons, a11y, etc addons are working.
- Chromatic changes should be gone on this PR.
